modules/core/native-system/src/main/java/org/rhq/core/system/ProcessExecutionOutputStream.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * RHQ Management Platform
+ * Copyright (C) 2005-2015 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2, as
+ * published by the Free Software Foundation, and/or the GNU Lesser
+ * General Public License, version 2.1, also as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.rhq.core.system;
+
+import java.io.ByteArrayOutputStream;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * in-memory output stream with a limited buffer size. Once limit is reached it starts to ignore
+ * {@link #write(int)} and {@link #write(byte[], int, int)} calls. Default limit is 2MB - this default can be
+ * set using  rhq.process-execution.captured-output.limit system property (in Bytes) or via {@link #setLimit(int)}.
+ * Optionally written output can be forwarded to logging subsystem - enable by setting {@link #writeToLog} to true.
+ * @author lzoubek@redhat.com
+ *
+ */
+public class ProcessExecutionOutputStream extends ByteArrayOutputStream {
+
+    private static final Log LOG = LogFactory.getLog(ProcessExecutionOutputStream.class);
+
+    private static final int DEFAULT_LIMIT;
+
+    static {
+        int limit = 2 * 1024 * 1024; // default 2MB
+        try {
+            limit = Integer.parseInt(System.getProperty("rhq.process-execution.captured-output.limit",
+                String.valueOf(limit)));
+        } catch (NumberFormatException nfe) {
+        }
+        DEFAULT_LIMIT = limit;
+    }
+
```

Comment:
I think a call to `java.lang.Integer#getInteger(java.lang.String, int)` would be easier to read (and does the same thing).
```java
private static final int DEFAULT_LIMIT = Integer.getInteger("rhq.process-execution.captured-output.limit", 2 * 1024 * 1024);  // default 2MB
```